crates/store/re_protos/proto/rerun/v0/redap.proto

```diff
@@ -0,0 +1,275 @@
+    rerun.common.v0.Schema schema = 1;
+}
+
+message UpdateResourcePropertiesRequest {
+    CatalogEntry entry = 1;
+
+    // This dataframe must contain a column indicating the resource_id
+    rerun.common.v0.DataframePart properties = 2;
+}
+
+message UpdateResourcePropertiesResponse {}
+
+message GetResourcePropertiesRequest {
+    CatalogEntry entry = 1;
+
+    // Column projection - define which columns should be returned.
+    // Providing it is optional, if not provided, all columns should be returned
+    rerun.common.v0.ColumnProjection column_projection = 2;
+
+    // Filter specific recordings that match the criteria (selection)
+    rerun.common.v0.PropertiesFilter filter = 3;
+}
+
+message CreateIndexRequest {
+    CatalogEntry entry = 1;
+
+    // what kind of index do we want to create and what are
+    // its index specific properties
+    rerun.common.v0.IndexProperties properties = 2;
+    // Component / column we want to index
+    rerun.common.v0.IndexColumn column = 3;
+    // What is the filter index i.e. timeline for which we
+    // will query the timepoints
+    // TODO(zehiko) this might go away and we might just index
+    // across all the timelines
+    rerun.common.v0.IndexColumnSelector time_index = 4;
+}
+
+message CreateIndexResponse {
+    uint64 indexed_resources = 1;
+    uint64 indexed_rows = 2;
+}
+
+message SearchIndexRequest {
+    CatalogEntry entry = 1;
+
+    // Index column that is queried
+    rerun.common.v0.IndexColumn column = 2;
+    // Query data - type of data is index specific. Caller must ensure
+    // to provide the right type. For vector search this should
+    // be a vector of appropriate size, for inverted index this should be a string.
+    // Query data is represented as a unit (single row) RecordBatch with 1 column.
+    rerun.common.v0.DataframePart query = 3;
+    // Index type specific properties
+    rerun.common.v0.IndexQueryProperties properties = 4;
+    // max number of rows to be returned
+    optional uint32 limit = 5;
+}
+
+message RegisterResourcesRequest {
+    CatalogEntry entry = 1;
+
+    // This has required columns based on the handler of the particular managed catalog entry
+    rerun.common.v0.DataframePart resource = 4;
+}
+
+message UnregisterResourcesRequest {}
+message UnregisterResourceResponse {}
+
+message UnregisterAllResourcesRequest {}
+message UnregisterAllResourcesResponse {}
+
+// ---------------- Resource APIs ------------------
+
+message GetResourceSchemaRequest {
+    rerun.common.v0.RecordingId recording_id = 1;
+}
+
+message GetResourceSchemaResponse {
+    rerun.common.v0.Schema schema = 1;
+}
+
+message GetChunkInfosRequest {
+    rerun.common.v0.RecordingId recording_id = 1;
+
+    // TODO: Chunk-filtering options
+    // Time-range
+    // Entity-paths
+}
+
+message GetChunkRequest {
+    rerun.common.v0.RecordingId recording_id = 1;
+    rerun.common.v0.Tuid chunk_id = 2;
+}
```

Comment:
```suggestion
message GetChunksRequest {
    rerun.common.v0.RecordingId recording_id = 1;
    repeated rerun.common.v0.Tuid chunk_ids = 2;
}
```

Reply:
We also need a way to stream chunks to the resources to replace `WriteMessages`.